### PR TITLE
Use toString for enum serialization

### LINKED ISSUE #2219: Adding an Enum in Breadcrumb setData causes a weird behaviour in serialization leading to a huge heap space allocation
### Integration

sentry

### Java Version

11

### Version

6.3.1

### Steps to Reproduce

- Create an enum with a few entries (local environment took 8 entries to cause a JVM crash)
- Add a breadcrumb (any level, any message) with said enum in the value of `setData`
- Send any event to sentry (making it serialize all the breadcrumbs)

From my tests is seems the problem happens in `JsonReflectionObjectSerializer` class at line 77: `Field[] fields = object.getClass().getDeclaredFields()`
Calling this on a enum returns all other enums making the serialization size and time increase exponentially.

Code example:
```kotlin
enum class DummyEnum {
        E1, E2
    }

@Test
    fun sentrySerializerTest() {
        val sw = StringWriter()

        val jsonWriter = JsonObjectWriter(sw, 100)

        val dummyLogger = object: ILogger {
            override fun log(level: SentryLevel, message: String, vararg args: Any?) { }
            override fun log(level: SentryLevel, message: String, throwable: Throwable?) { }
            override fun log(level: SentryLevel, throwable: Throwable?, message: String, vararg args: Any?) { }
            override fun isEnabled(level: SentryLevel?): Boolean { return false }
        }

        val duration = measureTime {
            jsonWriter.value(dummyLogger, DummyEnum.E1)
            println(sw.toString())
        }

        println(duration.inWholeMilliseconds)
    }
```

Running this piece of code prints out:

```
{"E1":"E1","E2":{"E1":"E1","E2":"E2","$VALUES":["E1","E2"]},"$VALUES":["E1",{"E1":"E1","E2":"E2","$VALUES":"[Lcom.mypackage.MyTestClass$DummyEnum;@4647881c"}]}
```

Adding more entries to DummyEnum increases the string size, time to process and memory exponentially

### Expected Result

Serializing an enum should be just the enum name.

### Actual Result

A lot of investigation is in the Steps to reproduce, but the actual result of the enum serialization in the breadcrumb is a multiplication of all enum entries:

```
{"E1":"E1","E2":{"E1":"E1","E2":"E2","$VALUES":["E1","E2"]},"$VALUES":["E1",{"E1":"E1","E2":"E2","$VALUES":"[Lcom.mypackage.MyTestClass$DummyEnum;@4647881c"}]}
```

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Use `toString()` for enum entries being serialized.

## :bulb: Motivation and Context

`JsonReflectionObjectSerializer` would consume lots of memory when serializing an enum entry where the enum has more than just a few entries.

Fixes #2219 

## :green_heart: How did you test it?
Unit Test

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

**Comment by rsolci** [comment]
Hey @adinauer, really thanks for the quick response. But I am thinking if this solution wouldn't be better in the `JsonObjectSerializer` class since there is already a conditional checking the type of the object to serialize and only calls the reflection one as a fallback for unknown types.

**Comment by adinauer** [comment]
> better in the JsonObjectSerializer class since there is already a conditional checking the type of the object to serialize and only calls the reflection one as a fallback for unknown types.

As an optimmization it could be added to both. But it should be in `JsonReflectionObjectSerializer` to also cover fields of complex objects being serialized.

File: sentry/src/main/java/io/sentry/JsonObjectSerializer.java [hunk 1]
```diff
@@ -46,6 +46,8 @@ public void serialize(
       serializeCollection(writer, logger, Arrays.asList((Object[]) object));
     } else if (object instanceof Map) {
       serializeMap(writer, logger, (Map<?, ?>) object);
+    } else if (object.getClass().isEnum()) {
+      writer.value(object.toString());
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/JsonObjectSerializer.java:50]: WHat's about `object..name()`? Does it return the same thing?

Comment (adinauer) [reply sentry/src/main/java/io/sentry/JsonObjectSerializer.java:50]: We'd have to use reflection to invoke `.name()`. By default `.toString()` should be the same as `.name()`. It also allows overriding the value that is serialized.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**